pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -213,7 +213,18 @@ func (i *KumaInjector) InjectKuma(ctx context.Context, pod *kube_core.Pod) error
 		if err != nil {
 			return err
 		}
-		pod.Spec.InitContainers = append(pod.Spec.InitContainers, patchedIc)
+		enabled, exist, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaInitFirst)
+		if err != nil {
+			return err
+		}
+		var initContainers []kube_core.Container
+		if exist && enabled {
+			log.V(1).Info(`injecting kuma init container first because kuma.io/init-first is set`)
+			initContainers = append([]kube_core.Container{patchedIc}, pod.Spec.InitContainers...)
+		} else {
+			initContainers = append(pod.Spec.InitContainers, patchedIc)
+		}
+		pod.Spec.InitContainers = initContainers
```

Comment:
I don't think this intermediate variable is necessary tbh
```suggestion
		if exist && enabled {
			log.V(1).Info("injecting kuma init container first because kuma.io/init-first is set")
			pod.Spec.InitContainers = append([]kube_core.Container{patchedIc}, pod.Spec.InitContainers...)
		} else {
			pod.Spec.InitContainers = append(pod.Spec.InitContainers, patchedIc)
		}
```